juju/client/proxy/kubernetes/proxy.py

```diff
@@ -0,0 +1,76 @@
+import tempfile
+
+from juju.client.proxy.proxy import Proxy, ProxyNotConnectedError
+from kubernetes import client
+from kubernetes.client.models.v1_label_selector import V1LabelSelector
+from kubernetes.stream import portforward
+
+class KubernetesProxy(Proxy):
+    def __init__(
+            self,
+            api_host,
+            namespace,
+            remote_port,
+            service,
+            service_account_token,
+            ca_cert=None,
+    ):
+        config = client.Configuration()
+        config.host = api_host
+        config.ssl_ca_cert = ca_cert
+        config.api_key = {"authorization": "Bearer " + service_account_token}
+
+        self.namespace = namespace
+        self.remote_port = remote_port
+        self.service = service
+
+        try:
+            self.remote_port = int(remote_port)
+        except ValueError:
+            raise ValueError("Invalid port number: %s" % remote_port)
```

Comment:
You would usually just use `remote_port` directly, and the caller would have to pass in and int or int-like object. Is there a reason this should do the string parsing in this class?

Reply:
Yeah ports aren't integers. No maths is performed on them. It's theoretically possible that we may want to open this up in the future to support named ports on the Kubernetes side.